src/bootstrap/cache.rs

```diff
@@ -95,7 +95,7 @@ impl<T: Internable + Deref> Deref for Interned<T> {
     }
 }
 
-impl<T: Internable + AsRef<U>, U: ?Sized> AsRef<U> for Interned<T> {
+impl<T: Internable + AsRef<U>, U: ?Sized + 'static> AsRef<U> for Interned<T> {
     fn as_ref(&self) -> &'static U {
```

Comment:
I would expect line 94 just above is similarly unsound, but I'm not sure if that is actually true.

(At least given the current listed of Internable impls)

We never move or mutate the buffers we're pointing to, so I think 'static here is mostly "right".

(Maybe modulo using with the non singleton types)

In any case this seems fine as a PR, r=me.

Reply:
Actually, neither of those `'static` lifetimes are doing us either good, since we can't ever take advantage of them from the  caller. Relaxed the PR a bit.